### ISSUE TITLE
Just placing an AudioStreamPlayer2D in a scene logs an error ingame

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#20235]
Godot 3.0.5 and master f1970e15b9450bff321f854d5ff1dbfc4bf80148
Windows 10 64 bits

Repro steps:

1) Create a new scene and place an `AudioStreamPlayer2D` in it. Leave the stream empty. By default, options show it should not play anyways:

![image](https://user-images.githubusercontent.com/1311555/42853887-0d00d16c-8a31-11e8-8126-7c9cfa93027a.png)

2) Play the scene: notice an error is printed in the console:
```
ERROR: set_stream: Condition ' !p_stream.is_valid() ' is true.
   At: scene/2d/audio_stream_player_2d.cpp:244
```

This only happens with `AudioStreamPlayer2D` and `AudioStreamPlayer3D`.

I am also getting this as part of #20183.

**Comment by vnen** [comment]
This was fixed in master by #19646. Needs to be cherry-picked to the stable branch.

**Comment by akien-mga** [comment]
Closing as duplicate of #16846 then.
